crates/bevy_ecs/src/world/error.rs

```diff
@@ -1,12 +1,62 @@
-//! Contains error types returned by bevy's schedule.
+//! Contains error types returned by methods on [`World`].
+//!
+//! [`World`]: crate::world::World
+
+use std::any::TypeId;
 
 use thiserror::Error;
 
-use crate::schedule::InternedScheduleLabel;
+use crate::{component::ComponentId, prelude::*, schedule::InternedScheduleLabel};
 
 /// The error type returned by [`World::try_run_schedule`] if the provided schedule does not exist.
 ///
 /// [`World::try_run_schedule`]: crate::world::World::try_run_schedule
 #[derive(Error, Debug)]
 #[error("The schedule with the label {0:?} was not found.")]
 pub struct TryRunScheduleError(pub InternedScheduleLabel);
+
+/// The error type returned by [`World::get_reflect`] and [`World::get_reflect_mut`].
+///
+/// [`World::try_run_schedule`]: crate::world::World::try_run_schedule
+#[cfg(feature = "bevy_reflect")]
+#[derive(Error, Debug)]
+pub enum GetComponentReflectError {
+    /// There is no [`ComponentId`] corresponding to the given [`TypeId`].
+    ///
+    /// Did you call [`App::register_type()`]?
+    ///
+    /// [`App::register_type()`]: ../../../bevy_app/struct.App.html#method.register_type
```

Comment:
You should be able to link to the method like:
```rust
    /// [`App::register_type`]: crate::app::App::register_type
```

Note that I believe we tend to leave off the `()` when referencing functions/methods.

Same for the `MissingReflectFromPtrTypeData` variant.

Reply:
Removed the `()` (everywhere I could find).

So, about the linking for `App::register_type`.. I started out with something like `crate::app::App::register_type`, but that doesn't work:

```
warning: unresolved link to `crate::app::App::register_type`
  --> crates/bevy_ecs/src/world/error.rs:29:33
   |
29 |     /// [`App::register_type`]: crate::app::App::register_type
   |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no item named `app` in module `bevy_ecs`
```

I think the problem is that `bevy_ecs` does not depend on `bevy_app` (but the other way around), so rustdoc has no chance to find `App`. That's why I went with the manual file-path link.

That said, I just tried out `crate::app::App::register_type` again, because I really would like a better solution for this, but it doesn't work, with the error given above.

Is it possible to declare like a "doc-dependency" on `bevy_app`, without introducing a cycle? :sweat_smile: I think not, but I feel like it could work in theory, but probably not in rustdoc/cargo doc for now.

Reply:
Oh that makes sense. In that case I might suggest linking to `TypeRegistry::register` directly or just not making `App::register_type` a link at all. I don't know I we ever link to crate dependents with a relative rustdoc path (I also don't know if that works for docs.rs docs).

@alice-i-cecile any thoughts here?

Reply:
I guess we can check what's gonna happen on docs.rs? Shouldn't break anything anyways.